### PR TITLE
Fix TSDB issues on Windows, add CI

### PR DESCRIPTION
**Pull request by brian-brazil**

### CODE REVIEW

**Comment by roidelapluie** [comment]
@brian-brazil does this need to be added to github repo config as well?

**Comment by brian-brazil** [comment]
This is me playing around, we don't have the right sort of CircleCI account for it though.

**Comment by roidelapluie** [comment]
you need to add it to jobs: as well

File: .circleci/config.yml [hunk 1]
```diff
@@ -49,6 +50,13 @@ jobs:
         key: v1-npm-deps-{{ checksum "web/ui/react-app/yarn.lock" }}
         paths:
         - web/ui/react-app/node_modules
+  test_windows:
+    executor: win/default
+    working_directory: /go/src/github.com/prometheus/prometheus
+    steps:
+    - checkout
+    - run:
+      - command: go test ./...
```
Comment (roidelapluie) [inline .circleci/config.yml]: shell: bash.exe

File: .circleci/config.yml [hunk 2]
```diff
@@ -49,6 +50,12 @@ jobs:
         key: v1-npm-deps-{{ checksum "web/ui/react-app/yarn.lock" }}
         paths:
         - web/ui/react-app/node_modules
+  test_windows:
+    executor: win/default
+    working_directory: /go/src/github.com/prometheus/prometheus
+    steps:
+    - checkout
+    - run: go test ./...
```
Comment (roidelapluie) [inline .circleci/config.yml]:
```suggestion
    - run:
        command: go test ./...
        shell: bash.exe
```

Comment (roidelapluie) [reply .circleci/config.yml]: mmh it works with powershell apparently.

File: tsdb/index/index.go [hunk 3]
```diff
@@ -720,6 +720,7 @@ func (w *Writer) writePostingsOffsetTable() error {
 	}
 
 	// Cleanup temporary file.
+	f.Close() // Needed for Windows.
```
Comment (bwplotka) [inline tsdb/index/index.go]: Just curious.. why? (:

Comment (bwplotka) [reply tsdb/index/index.go]: :+1: 

f is mmap of fP0 so it's important to close this handler. I would say it's sane for non windows as well (:

File: tsdb/index/index.go [hunk 4]
```diff
@@ -962,9 +963,9 @@ type labelIndexHashEntry struct {
 }
 
 func (w *Writer) Close() error {
-	if err := w.ensureStage(idxStageDone); err != nil {
-		return err
-	}
+	// Even if this failes, we need to close all the files.
```
Comment (roidelapluie) [inline tsdb/index/index.go]:
```suggestion
	// Even if this fails, we need to close all the files.
```

Comment (brian-brazil) [reply tsdb/index/index.go]: Fixed

**Comment by brian-brazil** [comment]
@simonpasquier @SuperQ This is touching CI, so one of ye should have a look at that bit. I did just enough to get it working.

**Comment by brian-brazil** [comment]
This is ready for review.

File: tsdb/index/index.go [hunk 5]
```diff
@@ -720,6 +724,8 @@ func (w *Writer) writePostingsOffsetTable() error {
 	}
 
 	// Cleanup temporary file.
+	f.Close() // Needed for Windows.
```
Comment (bwplotka) [inline tsdb/index/index.go]: Do we care about errors on close? (I know we used to not care) :thinking: should we log at least? It's a pretty unlikely error and not related to the bug we fix here, so not really necessary for this PR.

FYI we have this helper on Thanos: https://github.com/thanos-io/thanos/blob/865d5ec710b5651ead08a7dc73864a6f5dbeeaa9/pkg/runutil/runutil.go#L119

Comment (brian-brazil) [reply tsdb/index/index.go]: We don't generally care, but I've returned it on the happy path here just in case.

Comment (bwplotka) [reply tsdb/index/index.go]: Thanks!

**Pull request merged**